### DISCUSSION TITLE
Mcp23017 interrupt

### DISCUSSION THREAD

**Discussion by salvodjkr** [tttapa/Control-Surface#1020] [General]
Hi everyone . technical question. is it possible to use only one interrupt for two mcp23017?

**Comment by tttapa** [comment]
Yes. The interrupt pins are configured in open-drain mode:

https://github.com/tttapa/Control-Surface/blob/344324d0e662e2a4c2b1a7336d078f14c5e13c16/src/AH/Hardware/ExtendedInputOutput/MCP23017.ipp#L83-L93

**Comment by salvodjkr** [comment]
Hi Pieter, thanks for the immediate response. another question . given the complexity I decided to use a single interrupt, I have no skills to modify my code. what maximum distance can I use for the interrupt cable??.

**Reply by tttapa** [reply]
This will depend on the strength of the pull-up resistor you're using, and on the noise in the environment. You're much more likely to be limited by the length of the I²C wires: it's not meant to be used over distances longer than a couple of centimeters, maybe 30cm at most.
